ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,816 @@
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @param {String} name name of the channel
+         * @param {Bool} isPrivate true for the authenticated url, false for the public url
+         * @param {String|undefined} symbol is required for all public channels, not required for private channels (except position)
+         * @param {Object} params extra parameters specific to the poloniex api
+         * @returns {Object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        let token = undefined;
+        if (isPrivate) {
+            token = await this.getPrivateToken ();
+        } else {
+            token = await this.getPublicToken ();
+        }
+        const url = this.urls['api']['ws'] + '?token=' + token;
+        const milliseconds = this.numberToString (this.milliseconds ());
+        const subscribe = {
+            'id': milliseconds + name + symbol,   // ID is a unique string to mark the request which is same as the id property of ack.
+            'type': 'subscribe',
+            'topic': name,                                // Subscribed topic. Some topics support subscribe to the data of multiple trading pairs through ",".
+            'privateChannel': isPrivate,                  // Adopt the private channel or not. Set as false by default.
+            'response': true,                             // Whether the server needs to return the receipt information of this subscription or not. Set as false by default.
+        };
+        let messageHash = name;
+        if (symbol !== undefined) {
+            const market = this.market (symbol);
+            const marketId = market['id'];
+            messageHash = name + ':' + marketId;
+            subscribe['topic'] = messageHash;
+        }
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
+    }
+
+    async watchTicker (symbol: string, params = {}) {
+        /**
+         * @method
+         * @name poloniexfutures#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://futures-docs.poloniex.com/#get-real-time-symbol-ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the poloniexfutures api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const name = '/contractMarket/ticker';
+        return await this.subscribe (name, false, symbol, params);
+    }
+
+    async watchTrades (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}) {
```

Comment:
watchTrades does not seem to be working I believe it is subscribing to `/contractMarket/executionBTC/USDT:USDT` instead of `/contractMarket/execution:BTCUSDTPERP`

Reply:
are you sure you're not looking at the `id`? The `topic` is correct

```
2023-05-04T01:02:10.229Z sending {
  id: '1683162128882/contractMarket/executionBTC/USDT:USDT',
  type: 'subscribe',
  topic: '/contractMarket/execution:BTCUSDTPERP',
  privateChannel: false,
  response: true
}
2023-05-04T01:02:10.233Z onMessage { id: 'hE1vLHB7Oy', type: 'welcome' }
2023-05-04T01:02:10.594Z onMessage {
  id: '1683162128882/contractMarket/executionBTC/USDT:USDT',
  type: 'ack'
}
2023-05-04T01:02:13.560Z onMessage {
  data: {
    makerUserId: '1410880',
    symbol: 'BTCUSDTPERP',
    sequence: 799891,
    side: 'sell',
    size: 1,
    price: 29031.1,
    takerOrderId: '645304150288440007317082',
    makerOrderId: '6453041500324300072e347a',
    takerUserId: '1410880',
    tradeId: '64530415de029f00014795c5',
    ts: 1683162133640485000
  },
  subject: 'match',
  topic: '/contractMarket/execution:BTCUSDTPERP',
  type: 'message'
}
id |        symbol | timestamp | datetime | order | type | side | takerOrMaker |   price | amount | cost | fee | fees
---------------------------------------------------------------------------------------------------------------------
   | BTC/USDT:USDT |           |          |       |      |      |        taker | 29031.1 |        |      |  {} | [{}]
1 objects
```